[Eugene Lim] Duke Increments

eugenood opened the pull request.

j-lum committed:
Add support for Gradle workflow

j-lum committed:
Add sample checkstyle configuration

Add toolVersion block in to Gradle code sample to prevent errors.

Commit:
Add features for Level-1

Commit:
Add features for Level-2

Commit:
Add features for A-Classes

Commit:
Add features for Level-3

Commit:
Add features for A-Inheritance

Commit:
Add features for Level-4

Commit:
Add features for A-Exceptions

Commit:
Add features for Level-5

Commit:
Add features for Level-6

j-lum committed:
Change file mode on `gradle` to be executable

j-lum committed:
Merge pull request #12 from j-lum/gradle+x

Change file mode on `gradle` to be executable (#9)

j-lum committed:
Add configuration for console applications

Gradle defaults to an empty stdin which results in runtime exceptions
when attempting to read from `System.in`. Let's add some sensible
defaults for students who may still need to work with the standard
input stream.

j-lum committed:
Merge pull request #13 from j-lum/javaexec

Add configuration for console applications

Commit:
Add features for Level-7

Commit:
Add features for Level-8

Commit:
Merge branch 'branch-Level-7'

Commit:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Duke.java

Commit:
Fix merge errors for Level-8

Commit:
Refactor code to extract out closely related code as classes

Commit:
Organize the classes into suitable java packages

Commit:
Add JUnit test to Event class

Commit:
Merge branch 'gradle'

Commit:
Add Gradle support

Commit:
Add shadow plugin for JAR building

Commit:
Tweak the code to comply with a chosen coding standard

Commit:
Add JavaDoc comments

Commit:
Add features for Level-9

Commit:
Merge branch 'branch-A-CodingStandard'

Commit:
Merge branch 'branch-A-JavaDoc'

Commit:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/duke/Parser.java

Commit:
Add JavaFX GUI

madanalogy commented:
Hi, I really like how OOP your code is, makes it easy to understand the different components and follow along. Quite cool also how your taskList just extended off ArrayList rather than containing an ArrayList, I never thought of that, haha. I don't really have any suggestions sorry, great work! :)

eugenood commented:
Thanks @madanalogy. Appreciate your feedback :satisfied:

danielwys commented:
Good job! I admire that you've already completed the JavaFX portion of Duke so much in advance, and that you've accomplished all the levels, even the optional ones. Your codestyle and formatting is also consistent throughout. It was a pleasure to review your code.

Commit:
Rewrite way too many things

This is a bad commit. Don't do this in tP.
Essentially, I did the following:

1. Refactor for better OOP.
2. Add more exceptions handling.
3. Reorder package.
4. Change UI design.

Commit:
Merge pull request #5 from nexolute/branch-A-CodeQuality

Rewrite way too many things

Commit:
Add assertion

Commit:
Add friendlier syntax

Commit:
Add travis support

Commit:
Fix checkstyle complains

Commit:
Change profile picture

Commit:
Add dark mode

Commit:
Add screenshot to README

thaowei commented on src/main/java/duke/exception/DukeException.java:
```diff
@@ -0,0 +1,7 @@
+
+public abstract class DukeException extends Exception {
+
+    public abstract String getResponseMessage();
```
I feel like an exception class shouldn't be responsible for what is displayed to the user.

thaowei commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,39 @@
+        return this.description;
+    }
+
+    public abstract String toEncodedString();
```
Your tasks are responsible for encoding themselves but there's a separate parser class.

Commit:
Add user guide.

Commit:
Merge pull request #7 from nexolute/branch-A-UserGuide

Add user guide.

Commit:
Fix JavaFX warning

Commit:
Modify gradle to fix cross-platform compilation

Commit:
Change javafx version in fxml

Commit:
Fix Javafx cross platform bug
